catkit/hardware/flir/flir_camera.py

```diff
@@ -0,0 +1,356 @@
+        the PySpin Camera object.
+    read_only : boolean
+        Whether to make the property read-only or not.
+
+    Returns
+    -------
+    property
+        The created Python property.
+    '''
+    def getter(self):
+        return getattr(self.cam, flir_property_name).GetValue()
+
+    if read_only:
+        setter = None
+    else:
+        def setter(self, value):
+            getattr(self.cam, flir_property_name).SetValue(value)
+
+    return property(getter, setter)
+
+
+def _create_enum_property(flir_property_name, enum_name):
+    '''Helper for creating enum PySpin properties.
+
+    Parameters
+    ----------
+    flir_property_name : string
+        The name of the PySpin property. This property must be accessible from
+        the PySpin Camera object.
+    enum_name : dictionary
+        A dictionary describing the mapping from string (key) to PySpin enum value (value).
+
+    Returns
+    -------
+    property
+        The created Python property.
+    '''
+    def getter(self):
+        value = getattr(self.cam, flir_property_name).GetValue()
+
+        # Reverse search in enum dictionary.
+        for key, val in getattr(self, enum_name).items():
+            if value == val:
+                return key
+
+        raise KeyError('Value not recognized.')
+
+    def setter(self, value):
+        value = getattr(self, enum_name)[value]
+        getattr(self.cam, flir_property_name).SetValue(value)
+
+    return property(getter, setter)
+
+
+class FlirCamera(Camera):
+    instrument_lib = FlirLibrary
+
+    _system = None
+    _system_ref_count = 0
+
+    @classmethod
+    def _create_system(cls):
+        '''Create the PySpin system instance and keep ref count for tidy release.
+        '''
+        if cls._system is None:
+            cls._system = cls.instrument_lib.System.GetInstance()
+
+        cls._system_ref_count += 1
+
+    @classmethod
+    def _destroy_system(cls):
+        '''Destroy our reference to the PySpin system instance and release it if there
+        are no references to it anymore.
+        '''
+        cls._system_ref_count -= 1
+
+        if cls._system_ref_count == 0:
+            cls._system.ReleaseInstance()
+            cls._system = None
+
+    def initialize(self):
+        self.log = logging.getLogger(__name__)
+
+        # Set up enum dictionaries
+        self.pixel_format_enum = {'mono8': self.instrument_lib.PixelFormat_Mono8,
+                                  'mono12p': self.instrument_lib.PixelFormat_Mono12p,
+                                  'mono16': self.instrument_lib.PixelFormat_Mono16}
+
+        self.adc_bit_depth_enum = {'8bit': self.instrument_lib.AdcBitDepth_Bit8,
+                                   '10bit': self.instrument_lib.AdcBitDepth_Bit10,
+                                   '12bit': self.instrument_lib.AdcBitDepth_Bit12,
+                                   '14bit': self.instrument_lib.AdcBitDepth_Bit14}
```

Comment:
In case you read this and make alterations to un-name-mangle the "enum" before reading on, I would read on. I really get the impression that you just wanna use an ``Enum``.